webapp/security/views.py

```diff
@@ -333,5 +342,182 @@ def cve(cve_id):
+        return flask.jsonify({"message": error}), 400
+
+    return flask.jsonify({"message": "CVE created succesfully"}), 201
+
+
+def update_cve(cve_id):
+    data = flask.request.json
+    cve = db_session.query(CVE).filter(CVE.id == cve_id).first()
+
+    # Check if CVE exists by candidate
+    if len(db_session.query(CVE).all()) == 0:
+        response = flask.jsonify({"message": "CVE does not exist"}), 400
+        return response
+
+    cve.status = data.get("status")
+    cve.last_updated_date = data.get("last_updated_date")
+    cve.public_date_usn = data.get("public_date_usn")
+    cve.public_date = data.get("public_date")
+    cve.priority = data.get("priority")
+    cve.crd = data.get("crd")
+    cve.cvss = data.get("cvss")
+    cve.assigned_to = data.get("assigned_to")
+    cve.discovered_by = data.get("discovered_by")
+    cve.approved_by = data.get("approved_by")
+    cve.description = data.get("description")
+    cve.ubuntu_description = data.get("ubuntu_description")
+    cve.notes = data.get("notes")
+
+    if data.get("references"):
+        cve.references.clear()
+        for uri in data["references"]:
+            reference = (
+                db_session.query(CVEReference)
+                .filter(CVEReference.uri == uri)
+                .first()
+            )
+            if not reference:
+                reference = uri
+            cve.references.append(reference)
+
+    if data.get("bugs"):
+        cve.bugs.clear()
+        for b in data["bugs"]:
+            bug = db_session.query(Bug).filter(Bug.uri == uri).first()
+            if not bug:
+                bug = Bug(uri=b)
+            cve.bugs.append(bug)
+
+    # Packages
+    # Check if there are packages before mapping
+    if data.get("packages"):
+        cve.packages.clear()
+        for index, package_data in enumerate(data["packages"]):
+            package = Package(
+                name=package_data["name"],
+                source=package_data["source"],
+                ubuntu=package_data["ubuntu"],
+                debian=package_data["debian"],
+                releases=[],
+            )
+            package.releases.clear()
+            for release_data in package_data["releases"]:
+                package_release_status = PackageReleaseStatus(
+                    name=release_data["name"],
+                    status=release_data["status"],
+                    status_description=release_data["status_description"],
+                )
+
+                package.releases.append(package_release_status)
+            cve.packages.append(package)
+
+    try:
+        db_session.add(cve)
+        db_session.commit()
+    # Payload input errors
+    except exc.DataError as error:
+        return (
+            flask.jsonify({"message": error.orig.args[0]}),
+            400,
+        )
+    except exc.SQLAlchemyError as error:
+        return (
+            flask.jsonify({"message": error.args[0]}),
+            400,
+        )
```

Comment:
Similarly, in what cases are you expecting a `SQLAlchemyError` that isn't a `DataError`?

```suggestion
    except exc.DataError as error:
        return flask.jsonify({"message": error.orig.args[0]}), 400
```

Reply:
Data error is nearly always going to be about validation of the format of dates in this API. It may catch some other payload errors, but this is the only thing we have in the db that enforces a certain format.